File: src/main/java/net/andrecarbajal/naviMusic/commands/music/PlayCommand.java
package net.andrecarbajal.naviMusic.commands.music;

import lombok.extern.slf4j.Slf4j;
import net.andrecarbajal.naviMusic.audio.MusicService;
import net.andrecarbajal.naviMusic.commands.SlashCommand;
import net.andrecarbajal.naviMusic.util.URLUtils;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.interactions.commands.OptionMapping;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.OptionData;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class PlayCommand extends SlashCommand {
    private final MusicService musicService;

    public PlayCommand(MusicService musicService) {
        super("play", "Play track or playlist", Category.MUSIC);
        addOption(new OptionData(OptionType.STRING, "name-url-playlist", "Link or search query", true));
        this.musicService = musicService;
    }

    @Override
    public void onCommand(SlashCommandInteractionEvent event) {
        if (noVoiceChannelCheck(event)) return;
        event.deferReply().queue();
        try {
            String str = event.getOption("name-url-playlist", OptionMapping::getAsString);

            assert str != null;

            String PROVIDER = "ytsearch";
            
            if (URLUtils.isURL(str)) {
                if (str.toUpperCase().contains("SPOTIFY")) {
                    musicService.loadAndPlaySpotifyUrl(event.getChannel().asTextChannel(), PROVIDER, str, event.getMember()).editReply(event);
                    return;
                }
                musicService.loadAndPlayUrl(event.getChannel().asTextChannel(), str, event.getMember()).editReply(event);
            } else {
                musicService.loadAndPlay(event.getChannel().asTextChannel(), PROVIDER, str, event.getMember()).editReply(event);
            }
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/net/andrecarbajal/naviMusic/commands/music/PlayCommand.kt
package net.andrecarbajal.naviMusic.commands.music

import net.andrecarbajal.naviMusic.audio.MusicService
import net.andrecarbajal.naviMusic.commands.SlashCommand
import net.andrecarbajal.naviMusic.util.URLUtils
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.OptionData
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class PlayCommand(private val musicService: MusicService) :
    SlashCommand("play", "Play track or playlist", Category.MUSIC) {

    private val log = LoggerFactory.getLogger(PlayCommand::class.java)

    init {
        addOption(OptionData(OptionType.STRING, "name-url-playlist", "Link or search query", true))
    }

    override fun onCommand(event: SlashCommandInteractionEvent) {
        if (noVoiceChannelCheck(event)) return
        event.deferReply().queue()

        try {
            val query = event.getOption("name-url-playlist")?.asString ?: return
            val provider = "ytsearch"
            val textChannel = event.channel.asTextChannel()
            val member = event.member

            if (URLUtils.isURL(query)) {
                if (query.contains("spotify", ignoreCase = true)) {
                    musicService.loadAndPlaySpotifyUrl(textChannel, provider, query, member).editReply(event)
                } else {
                    musicService.loadAndPlayUrl(textChannel, query, member).editReply(event)
                }
            } else {
                musicService.loadAndPlay(textChannel, provider, query, member).editReply(event)
            }
        } catch (e: Exception) {
            log.error("Error in PlayCommand", e)
            throw RuntimeException(e)
        }
    }
}
